src/Engine/Tables/ItemTable.cpp

```diff
@@ -288,14 +288,20 @@ void ItemTable::LoadPotions(const Blob &potions) {
             return;
         }
         for (ItemId column : Segment(ITEM_FIRST_REAL_POTION, ITEM_LAST_REAL_POTION)) {
-            int flatPotionId = std::to_underlying(column) - std::to_underlying(ITEM_FIRST_REAL_POTION);
-            char *currValue = tokens[flatPotionId + 7];
-            potion_value = atoi(currValue);
-            if (!potion_value && currValue[0] == 'E') {
-                // values like "E{x}" represent damage level {x} when using invalid potion combination
-                potion_value = atoi(currValue + 1);
+            if (row == ITEM_POTION_SHIELD && column == ITEM_POTION_HARDEN_ITEM || column == ITEM_POTION_SHIELD && row == ITEM_POTION_HARDEN_ITEM) {
+                // Fix a vanilla bug that GrayFace fixes in a way OE doesn't see: Mixing a Potion of Water Resistance
+                potion_value = ITEM_POTION_WATER_RESISTANCE;
+            } else {
```

Comment:
You shouldn't need this specific exclusion - changing the type to ItemId will fix all

Reply:
I guess it may depend on what dataset you are using? Ive just checked the potion table from my GOG install and the entries are correct.